### PR TITLE
Add game functionality.

### PR DESCRIPTION
**Pull request by optimusinhisprime**
Add game functionality here.

### CODE REVIEW

File: tictactoe/src/components/Board.js [hunk 1]
```diff
@@ -1,16 +1,70 @@
...
-  //console.log(playerMovePositions);
+export default function Board({
+  playerMovePositions,
+  currentPlayer,
+  setCurrentPlayer,
+  setPlayerMovePosition,
+  isOver,
+  setGameOver,
+}) {
+  useEffect(() => {
```
Comment (ChinyangaTL) [inline tictactoe/src/components/Board.js]: The useEffect for the winner checking was a nice touch. 
Clean out the console log

File: tictactoe/src/components/Board.js [hunk 2]
```diff
@@ -1,16 +1,70 @@
...
+      alert("The game is over. Restart it to play again.");
+    }
+  }
+
+  function checkWinner() {
+    let winner = calculateWinner(playerMovePositions);
+    if (winner == "X") {
+      return winner;
+    } else if (winner == "O") {
+      return winner;
```
Comment (ChinyangaTL) [inline tictactoe/src/components/Board.js]: Could benefit from a tenery statement

```
const theWinner = winner === 'X' ? 'X' : 'O'
return theWinner
```

or alternatively

```
return winner === 'X' ? 'X' : 'O'
``` 
to have it as a oneliner

File: tictactoe/src/components/Game.js [hunk 3]
```diff
@@ -1,16 +1,47 @@
...
+          currentPlayer={currentPlayer}
+          setCurrentPlayer={setCurrentPlayer}
+          setPlayerMovePosition={setPlayerMovePosition}
+          setGameOver={setGameOver}
+        />
+
+        <button onClick={() => restartGame()}>Restart Game</button>
+      </div>
+    );
+  }
```
Comment (ChinyangaTL) [inline tictactoe/src/components/Game.js]: For conditional rendering

```
if(isOver) {
    return (<JSX/>)
}

if(anotherConditionIfYouHaveOne) {
    return (<JSX/>
}

// the base return
return (<JSX/>)
```

the else statement clogs things up and is generally uncessary cause the return will handle things.

**Review by ChinyangaTL — COMMENTED** [review]
Good stuff though. Functionality checks out. Syntax could use tightening up here and there, but I like what I am seeing.

File: tictactoe/src/components/Board.js (hunk 1 above)
Comment (optimusinhisprime) [reply tictactoe/src/components/Board.js]: Yea, lemmi get on that. Thanks man.

File: tictactoe/src/components/Game.js (hunk 3 above)
Comment (optimusinhisprime) [reply tictactoe/src/components/Game.js]: Did notice it on one of the features I was working on. I will refactor the code and reduce them.

File: tictactoe/src/components/Board.js (hunk 2 above)
Comment (optimusinhisprime) [reply tictactoe/src/components/Board.js]: Will consider using the new ES6 features when I clean up the code. Nice catch. These old ways are out here making me look like a boomer.

File: tictactoe/src/components/Board.js [hunk 4]
```diff
@@ -40,20 +40,37 @@ export default function Board({
...
+    });
+
+    if (result == null && counter === playerMovePositions.length) {
+      stopGame();
+      alert(
+        `Match drawn. “For every action, there was an equal and opposite reaction.”`
+      );
+    }
+  }
+
```
Comment (ChinyangaTL) [inline tictactoe/src/components/Board.js]: I see this 👏

File: tictactoe/src/components/Board.css [hunk 5]
```diff
@@ -1,9 +1,17 @@
 .board{
     display: inline-grid;
-    width: 300px;
-    height: 300px;
+    width: 250px;
+    height: 250px;
     grid-template-columns: 1fr 1fr 1fr;
     grid-template-rows: 1fr 1fr 1fr;
```
Comment (ChinyangaTL) [inline tictactoe/src/components/Board.css:6]:
```
grid-template-columns: repeat(3, 1fr) 
```
A bit more concise

File: tictactoe/src/components/GameOver.js [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import React from "react";
+import gameOver from "../images/gameOver.gif";
```
Comment (ChinyangaTL) [inline tictactoe/src/components/GameOver.js]: 😆😆 😆

**Comment by rmoswela** [comment]
Hey dude, improve the git commits message and nice to see you are slowly implementing what you learnt with git workflow

@ChinyangaTL nice of you to review. I hope to see @iseeflames also reviewing your code

Fix the conflicts due

the call me again to come merge

**Comment by optimusinhisprime** [comment]
Yea. I need to take a class on those at this point.  Thank You. 

@rmoswela done resolving the conflicts.

**Comment by optimusinhisprime** [comment]
@rmoswela please merge the PR.

File: README.md [hunk 7]
```diff
@@ -1,70 +1,15 @@
...
+### Game Rules
+1. The game is played on a grid that's 3 squares by 3 squares.
+2. You are X, your friend (or the computer in this case) is O. Players take turns putting their marks in empty squares.
+3. The first player to get 3 of her marks in a row (up, down, across, or diagonally) is the winner.
+4. When all 9 squares are full, the game is over.
 
-In the project directory, you can run:
+To run the game type:
+1. `npm install` to get all the dependencies
+2. `npm start` to run the game on the local server.
```
Comment (ChinyangaTL) [inline README.md:14]: I'd also suggest leaving a link to the deployed site.

**Pull request merged**